cpu/esp32/esp-now/esp_now_netdev.h

```diff
@@ -14,61 +14,95 @@
  * @brief       Netdev interface for the ESP-NOW WiFi P2P protocol
  *
  * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @author      Timo Rothenpieler <timo.rothenpieler@uni-bremen.de>
  */
 
 #ifndef ESP_NOW_NETDEV_H
 #define ESP_NOW_NETDEV_H
 
 #include "net/netdev.h"
+#include "ringbuffer.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
- * @brief   Maximum packet size that can be used with ESP-NOW
+ * @brief   Maximum raw packet size that can be used with ESP-NOW (including headers)
  */
-#define ESP_NOW_MAX_SIZE (250)
+#define ESP_NOW_MAX_SIZE_RAW (250)
 
 /**
  * @brief   Length of ESP-NOW addresses
  */
 #define ESP_NOW_ADDR_LEN ETHERNET_ADDR_LEN
 
+/**
+ * @brief   Size of non-data header elements in ESP-NOW packet
+ */
+#define ESP_NOW_HEADER_LENGTH (sizeof(esp_now_pkt_hdr_t))
+
+/**
+ * @brief   Maximum packet size that can be used with ESP-NOW
+ */
+#define ESP_NOW_MAX_SIZE (ESP_NOW_MAX_SIZE_RAW - ESP_NOW_HEADER_LENGTH)
+
+/**
+ * @brief   buffer size used for RX buffering
+ *
+ * Reduce this value if your expected traffic does not include full IPv6 MTU
+ * sized packets.
+ */
+#ifndef ESP_NOW_BUFSIZE
+#define ESP_NOW_BUFSIZE (1500)
+#endif
+
 /**
  * @brief   Reference to the netdev device driver struct
  */
 extern const netdev_driver_t esp_now_driver;
 
+/**
+ * @brief   Header with neccesary flags for ESP-NOW packets
+ */
+typedef struct __attribute__((packed))
```

Comment:
Should be placed in `esp_now_netif.h` since it is not related to the `esp_now` netdev and is only used in `esp_now_netif.h`. You could also think about to place it in `esp_now_netif.c` because it has not to be exposed for other modules. If you want to have it documented, it has to be placed in a header file of course.

Reply:
Its size is used in macros that are ultimately used to determine ESP_NOW_MAX_SIZE, which is used in various locations. So if it's moved to the netif header, the netdev header either needs to include it, or the users of ESP_NOW_MAX_SIZE need to include the netif header first.
I think it's best to just leave it where it is.